src/controllers/scripting/controllerscriptenginebase.h

```diff
@@ -25,7 +25,7 @@ class ControllerScriptEngineBase : public QObject {
 
     virtual bool initialize();
 
-    bool executeFunction(QJSValue functionObject, const QJSValueList& arguments);
+    bool executeFunction(QJSValue functionObject, const QJSValueList& arguments = {});
```

Comment:
I'm not sure if this has performance implications. Can you ensure this won't cause any problems because of the construction of a temporary or something? The QList default constructor does not give any guarantees.

Reply:
Doesn't matter. Instead of creating default arguments in various places a single default parameter suffices.

Reply:
The default instance would be created anyway, either here or at the call site.

Reply:
Nope, then your understanding is wrong. The substitution happens at compile time, either or.

But it wouldn't matter anyway, because the default constructor of all container types does almost nothing. QList is no exception.

Reply:
It is only syntactic sugar and for convenience, no performance implications.

Reply:
Ok thanks for the clarification.